test/functional/feature_taproot.py

```diff
@@ -495,11 +499,11 @@ def make_spender(comment, *, tap=None, witv0=False, script=None, pkh=None, p2sh=
 
     # Compute scriptPubKey and set useful defaults based on the inputs.
     if witv0:
-        assert tap is None
+        assert_equal(tap, None)
         conf["mode"] = "witv0"
         if pkh is not None:
             # P2WPKH
-            assert script is None
+            assert_equal(script, None)
```

Comment:
`==` and `is` are not entirely equivalent in Python, with both differences in evaluation speed and outcome. Even though I think it's fine to use the equality operator here, I think the proper way to do it would be to add assert_is and/or assert_is_none functions (e.g. unittest also has these: https://docs.python.org/3/library/unittest.html#assert-methods)

Reply:
I think I'm getting confused here :) 

We need to converge on some code style here otherwise we cause only confusion https://github.com/bitcoin/bitcoin/pull/23127#discussion_r784135049

cc @MarcoFalke